docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -116,9 +116,21 @@ use_pkce = true
 
 Grafana always uses the SHA256 based `S256` challenge method and a 128 bytes (base64url encoded) code verifier.
 
+### Configure refresh token
+
+> Available in Grafana v9.3 and later versions.
+
+> **Note:** This feature is behind the `accessTokenExpirationCheck` feature toggle.
+
+When a user logs in using an OAuth provider, Grafana verifies that the access token has not expired. When an access token expires, Grafana uses the provided refresh token (if any exists) to obtain a new access token.
+
+Grafana uses a refresh token to obtain a new access token without requiring the user to log in again. If a refresh token doesn't exist, Grafana logs the user out of the system after the access token has expired.
+
+Generic OAuth supports obtaining access tokens by using the refresh token, but the `[auth.generic_oauth]` section should be extended with additional scopes and it should be enabled on the provider side as well.
```

Comment:
This one sentence is saying a lot, wondering how we can divide this up a bit for clarity. How about:

"You can configure Generic OAuth to use a refresh token to obtain an access token by:
- Extending the `[auth.generic_oauth]` section with additional scopes
- Enabling it on the provider side."

What does 'it' refer to? I assume 'it' refers to 'Generic OAuth'?

Reply:
Agree, it is misunderstandable. In this case 'it' refers to 'refresh token'. 

There is an 'and/or' relation between __Extending the [auth.generic_oauth] section with additional scopes__ and
__Enabling it on the provider side.__  -> Sometimes either extending the auth.generic_oauth section with additional scopes or enabling it on the provider side is enough, sometimes both should be configured.

Reply:
Ah, ok, then we should explain the conditions under which they would perform those actions. Something along the lines of:

"- If this, then extend the auth.generic_oauth section with additonal scopes.
- If this other thing, then enable the refresh token on the provide.
-  If this third thing, do nothing"

Let's start with that. We could also consider adding a decision table with this content.

| If this... | Then perform the following |
| condition 1 | Steps |
| condition 2 | Steps |